src/GitHub/Lens.hs

```diff
@@ -0,0 +1,68 @@
+{-# LANGUAGE DataKinds     #-}
+{-# LANGUAGE PolyKinds     #-}
+{-# LANGUAGE TypeFamilies  #-}
+{-# LANGUAGE TypeOperators #-}
+
+{- |
+Copyright: (c) 2021 Kowainik
+SPDX-License-Identifier: MPL-2.0
+Maintainer: Kowainik <xrom.xkov@gmail.com>
+
+Utilities to provide general @lens@-like API for creating GraphQL
+queries.
+-}
+
+module GitHub.Lens
+    ( HasOwnerName (..)
+    , HasLimit (..)
+    , HasStates (..)
+
+      -- * Parameter types
+    , ArgsType (..)
+
+      -- * Internas
+    , Delete
+    ) where
+
+import Data.Kind (Type)
+import Data.List.NonEmpty (NonEmpty)
+import Data.Text (Text)
+import Prolens (Lens)
+
+import GitHub.GraphQL (State)
+
+
+{- | Type of fields for arguments.
+-}
+data ArgsType
+    = ArgsOwner
+    | ArgsName
+    | ArgsLimit
+    | ArgsStates
+
+{- | Typeclass for lenses that can change owner and name
```

Comment:
Should we specify somehow that they are both *always* should be together? Or if there will be situations later where only one of them is required, maybe it will make sense to separate this?